utils/replicaset/canary.go

```diff
@@ -163,7 +163,14 @@ func CalculateReplicaCountsForCanary(rollout *v1alpha1.Rollout, newRS *appsv1.Re
 	minAvailableReplicaCount := rolloutSpecReplica - MaxUnavailable(rollout)
 
 	totalAvailableOlderReplicaCount := GetAvailableReplicaCountForReplicaSets(oldRSs)
-	scaleDownCount := GetReplicasForScaleDown(newRS) + GetReplicasForScaleDown(stableRS) + totalAvailableOlderReplicaCount - minAvailableReplicaCount
+	stableRSReplicasForScaleDown := GetReplicasForScaleDown(stableRS)
+	scaleDownCount := GetReplicasForScaleDown(newRS) + stableRSReplicasForScaleDown + totalAvailableOlderReplicaCount - minAvailableReplicaCount
+
+	if scaleDownCount <= 0 && stableRSReplicasForScaleDown == 0 {
```

Comment:
Although this works for the scenario you provided, I don't think this is a complete fix, because  `stableRSReplicasForScaleDown == 0` is just one case and I think it would be possible for `stableRSReplicasForScaleDown > 0` and we would still face the problem. E.g. for the initial version, 1 out of 2 pods were available, but the other was CrashLoopBackoff-ing